### ISSUE TITLE
Nextcloud fails code integrity check

### ISSUE THREAD

**Issue by scintill** [NixOS/nixpkgs#192400]
### Describe the bug
Nextcloud instance fails Nextcloud's internal code integrity check. This produces a warning in Nextcloud's admin area.

### Steps To Reproduce
1. Enable nextcloud 24. E.g. `services.nextcloud = { enable = true; package = pkgs.nextcloud24; }`
2. Browse to the URL `/settings/admin/overview` on your nextcloud server.
3. You should see the red warning `Some files have not passed the integrity check. Further information on how to resolve this issue can be found in the documentation.`

### Screenshot
![image](https://user-images.githubusercontent.com/848492/191690443-1a97ddb3-6332-4775-bdee-756657b20217.png)

### Expected behavior
The warning should not appear; the integrity check should pass.

### Additional context
I believe this is caused by the patch in #190646.

### Notify maintainers
@schneefux @bachp @globin @ma27

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.68, NixOS, 22.05 (Quokka), 22.05.3167.e64df9c5c8f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(scintill): `""`
 - channels(root): `"nixos-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Ma27** [comment]
I think the patch is valid in our case, though (as described in the patch message).

The question is if there's something we can/should do about this warning or just leave it because it's a valid distro-level patch.

cc @aanderse @yayayayaka @lheckemann @nlewo  for opinions.

**Comment by yayayayaka** [comment]
It is a valid distro-level patch in my opinion. I think the best way to approach this is to get our patch merged upstream.

**Comment by scintill** [comment]
I agree, working with upstream would be best.

I found an undocument config.php option, `'integrity.check.disabled' => true` that got rid of the warning (you may have to re-run the scan, which is an option provided in the warning.) I recommend against disabling it this way in the official NixOS packaging, but maybe this is useful for individual users.

**Comment by BurNiinTRee** [comment]
The [Nextcloud docs](https://docs.nextcloud.com/server/24/admin_manual/issues/code_signing.html#not-open-source-anymore) suggest that custom distributions set the release version branch to something other than "stable" (maybe "stable-nix" or something in our case). IIUC this still just disables the checks as well.

**Comment by dotlambda** [comment]
Why don't we modify `$out/core/signature.json`?

**Comment by icewind1991** [comment]
Starting with nc26 you can disable the behavior that is currently patched out by setting `NC_setup_create_db_user=false` during the setup. So we should be able to have the check pass with nc26.

See https://github.com/nextcloud/server/pull/36428

**Comment by Luflosi** [comment]
This should be fixed for Nextcloud 26: #222372.
